File: completed/prob15.py
import math

def recursive(i, r, d, n):
    print "r = "+str(r)+"; d = "+str(d)
    if r==n or d==n:
        i+=1
        return i
    else:
        i = recursive(i, r+1, d, n)
        i = recursive(i, r, d+1, n)
        return i

n = int(raw_input("n = "))
print recursive(0, 0, 0, n)

File: completed/prob15.cpp
#include <iostream>
#include <math.h>

using namespace std;

unsigned long int recursive
         (unsigned long int i, int r, int d, int n)
{
  if (r==n || d==n)
    return ++i;
  else
  {
    i = recursive(i, r+1, d, n);
    i = recursive(i, r, d+1, n);
    return i;
  }
}

int main()
{
  unsigned long int i;
  int n;
  cout << "n = ";
  cin >> n;
  i = recursive(0, 0, 0, n);
  cout << i << endl;
  return 0;
}
